storage/disk/disk.go

```diff
@@ -357,7 +365,19 @@ func (db *Store) Truncate(ctx context.Context, txn storage.Transaction, params s
 
 	db.rmu.Unlock()
 
-	return db.cleanup(oldDb, backupDir)
+	err = db.cleanup(oldDb, backupDir)
+	if err != nil {
+		return wrapError(err)
+	}
+
+	symlink := filepath.Join(path.Dir(newDB.Opts().Dir), symlinkKey)
+	if _, err := os.Lstat(symlink); err == nil {
+		if err := os.Remove(symlink); err != nil {
+			return wrapError(err)
+		}
+	}
+
+	return wrapError(os.Symlink(newDB.Opts().Dir, symlink))
```

Comment:
Hmm if this goes wrong, for whatever reason, will we loose something? I.e. I'm wondering if we should do the symlink re-linking _before_ the cleanup; perhaps even before the swap? 🤔

Reply:
Good point. Moved the symlink re-linking before the db swap.